src/Storages/StorageDistributed.cpp

```diff
@@ -868,10 +844,13 @@ void StorageDistributed::rename(const String & new_path_to_table_data, const Sto
         renameOnDisk(new_path_to_table_data);
     renameInMemory(new_table_id);
 }
+
+
 void StorageDistributed::renameOnDisk(const String & new_path_to_table_data)
 {
-    for (const DiskPtr & disk : volume->getDisks())
+    for (const DiskPtr & disk : storage_policy->getDisks())
     {
+        disk->reserve(0);
```

Comment:
Why we need it now?

Reply:
I don't remember making these changes. @excitoon here should be `volume->getDisks` and remove `disk-reserve(0)`.

Reply:
Would it be better if I use `reserve()` from `volume`? @NanoBjorn 
This is https://github.com/ClickHouse/ClickHouse/pull/12670/commits/1b3f5c99f5917844f9643c46935279b85543e7b8

Reply:
I still don't understand why we needed to move back from volume as member to storage_policy, but maybe it is okay if it works. Why you need `reserve` here at all? It was fine without it, or not?